### PR TITLE
[ACS-6457] Migrated unshare-file e2es to playwright

### PR DESCRIPTION
**Pull request by kritagya09**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
  Protractor is deprecated and e2es for 'unshare-file.test.ts' are written in Protractor.

**What is the new behaviour?**
  Migrated the Protractor test cases into Playwright for 'unshare-file.test.ts' file.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
  https://alfresco.atlassian.net/browse/ACS-6457

### CODE REVIEW

File: projects/aca-playwright-shared/src/utils/timeouts.ts [hunk 1]
```diff
@@ -34,5 +34,5 @@ export const timeouts = {
   globalTest: 45 * 1000,
   extendedTest: 150 * 1000,
   webServer: 240 * 1000,
-  globalSpec: 60 * 10 * 1000
+  globalSpec: 60 * 20 * 1000
```
Comment (akashrathod28) [inline projects/aca-playwright-shared/src/utils/timeouts.ts:37]: can you please help to know, why this change required for these tests?

File: projects/aca-playwright-shared/src/utils/config.ts [hunk 2]
```diff
@@ -43,3 +43,13 @@ export const SIDEBAR_LABELS = {
   FAVORITES: 'Favorites',
   TRASH: 'Trash'
 };
+
+export const SITE_VISIBILITY = {
```
Comment (akashrathod28) [inline projects/aca-playwright-shared/src/utils/config.ts:47]: here we can use 'Site' import from '@alfresco/js-api' Site.VisibilityEnum.PRIVATE

File: projects/aca-playwright-shared/src/utils/config.ts [hunk 3]
```diff
@@ -43,3 +43,13 @@ export const SIDEBAR_LABELS = {
...
+  PRIVATE: 'PRIVATE'
+};
+
+export const SITE_ROLES = {
```
Comment (akashrathod28) [inline projects/aca-playwright-shared/src/utils/config.ts:51]: here we can use 'Site' import from '@alfresco/js-api' Site.RoleEnum.SiteConsumer

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+    await personalFiles.viewerDialog.shareDialogTitle.waitFor({ state: 'attached', timeout: timeouts.normal });
+
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+    await personalFiles.shareDialog.shareToggle.click();
+
+    expect(personalFiles.confirmDialog.getDialogTitle('Remove this shared link')).toBeVisible();
+    expect(
+      personalFiles.confirmDialog.getDialogContent('This link will be deleted and a new link will be created next time this file is shared')
+    ).toBeVisible();
+    expect(await personalFiles.confirmDialog.isRemoveEnabled()).toBe(true);
```
Comment (akashrathod28) [inline e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:130]: prefer [expect(locator).toBeEnabled()] to avoid flakiness

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    expect(await personalFiles.confirmDialog.isCancelEnabled()).toBe(true);
```
Comment (akashrathod28) [inline e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:131]: prefer [expect(locator).toBeEnabled()] to avoid flakiness

File: projects/aca-playwright-shared/src/utils/timeouts.ts (hunk 1 above)
Comment (kritagya09) [reply projects/aca-playwright-shared/src/utils/timeouts.ts:37]: I'm getting error in build i.e. timeout waiting 600s for the entire test run as below screenshots, that's why I increase global timeout.

Screenshot from 2023-12-26 16-04-38.

File: projects/aca-playwright-shared/src/utils/config.ts (hunk 2 above)
Comment (kritagya09) [reply projects/aca-playwright-shared/src/utils/config.ts]: Updated all suggested places.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    await personalFiles.shareDialog.shareToggle.click();
+
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(file4Id)).toBe(false);
+
+    await page.goto(url);
+    await personalFiles.viewer.waitForViewerToOpen();
+    expect(personalFiles.viewer.fileTitleButtonLocator).not.toEqual(file4);
+
+    await page.reload();
```
Comment (akashrathod28) [inline e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:179]: do we need this reload at end of script.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file4);
+
+    await personalFiles.dataTable.rightClickOnItem(file4);
+    await personalFiles.matMenu.clickMenuItem('Shared Link Settings');
+    const url = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(file4Id)).toBe(false);
```
Comment (akashrathod28) [inline e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:173]: we can move after Check on UI as we are getting flaky test due to backend take time to reflect file status.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file2);
+
+    await personalFiles.dataTable.selectItem(file2);
+    await personalFiles.acaHeader.shareButton.click();
+    const url = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(file2Id)).toBe(false);
```
Comment (akashrathod28) [inline e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:136]: we can move this line after Check on UI as we are getting flaky test due to backend take time to reflect file status.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(fileSite2);
+
+    await personalFiles.dataTable.selectItem(fileSite2);
+    await personalFiles.acaHeader.shareButton.click();
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+
+    await personalFiles.shareDialog.shareToggle.click();
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(fileSite2Id)).toBe(false);
```
Comment (akashrathod28) [inline e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:209]: we can add step to very on UI like right click on file menu should be 'Shared Link Settings',
or add some timeout before check on backend to avoid flaky test.

File: projects/aca-playwright-shared/src/utils/timeouts.ts (hunk 1 above)
Comment (akashrathod28) [reply projects/aca-playwright-shared/src/utils/timeouts.ts:37]: can you crate separate test suite for share-action. so each suite can execute under 10 mint. 
as we have action suite test in protractor split for e2e run.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts (hunk 6 above)
Comment (kritagya09) [reply e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts]: not needed, removed reload.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts (hunk 7 above)
Comment (kritagya09) [reply e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts]: Moved this line after check on UI, Thanks.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts (hunk 9 above)
Comment (kritagya09) [reply e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:208]: Added tiny timeout before check on backend, thanks!

File: projects/aca-playwright-shared/src/utils/timeouts.ts (hunk 1 above)
Comment (kritagya09) [reply projects/aca-playwright-shared/src/utils/timeouts.ts:37]: yes I can do but should I create test suit in same PR or need to raise a separate PR.

Comment (akashrathod28) [reply projects/aca-playwright-shared/src/utils/timeouts.ts:37]: you can do it in same.

File: e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+      file1Id = (await nodesApi.createFile(file1, parentId)).entry.id;
+      file2Id = (await nodesApi.createFile(file2, parentId)).entry.id;
+      file3Id = (await nodesApi.createFile(file3, parentId)).entry.id;
+      file4Id = (await nodesApi.createFile(file4, parentId)).entry.id;
+
+      await sitesApi.createSite(sitePrivate, Site.VisibilityEnum.PRIVATE);
+      const docLibId = await sitesApi.getDocLibId(sitePrivate);
+
+      fileSite1Id = (await nodesApi.createFile(fileSite1, docLibId)).entry.id;
+      fileSite2Id = (await nodesApi.createFile(fileSite2, docLibId)).entry.id;
```
Comment (swapnil-verma-gl) [inline e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:76]: Can probably move these 2 to local consts, as they are not getting reassigned anywhere

Comment (swapnil-verma-gl) [reply e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts:76]: Never mind, only applies to fileSite1Id. fileSite2Id is being used in tests

Comment (kritagya09) [reply e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts]: Updated.

File: e2e/playwright/share-action/src/tests/share/unshare-file-search-results.spec.ts [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { Site } from '@alfresco/js-api';
+import { ApiClientFactory, FileActionsApi, NodesApi, SharedLinksApi, SitesApi, Utils, test, timeouts } from '@alfresco/playwright-shared';
+import { expect } from '@playwright/test';
+
+test.describe.only('Unshare a file from Search Results', () => {
```
Comment (akashrathod28) [inline e2e/playwright/share-action/src/tests/share/unshare-file-search-results.spec.ts:29]: please remove .only

**Review by swapnil-verma-gl — APPROVED** [review]

**Review by akashrathod28 — APPROVED** [review]
